### ISSUE TITLE
Add option to disable and hide ‚update branch by merge‘

### ISSUE THREAD

**Issue by markusamshove** [go-gitea/gitea#21891]
### Feature Description

I’d like to propose to have a setting to disable the functionality to update a branch (head) with the base in a pull request and hide that action.

In our case we have pull requests targeting a QA branch which should never ever be merged into a feature branch. However, the button sounds too good to not click it sometimes :)

**Comment by phild5** [comment]
Yes, please. We don't want automatic updates by _merge_, only by _rebase_.
There is an option to disable automatic updates by rebase, but the additional option "by merge" can't explicitly be disabled.
![grafik](https://github.com/user-attachments/assets/ea8d6424-55aa-43f4-9121-51dcd965a0e2)
![grafik](https://github.com/user-attachments/assets/599e8652-01ce-43b3-8071-e905be99124d)

**Comment by Mo0rBy** [comment]
I would like to see the ability to disable all branch updates on the remote server (for a specific repo, not over the entire server) and only allow users to update their branches locally, then push the updated branch to the remote repository.

I don't want people to update by Rebase or Merge on the server because I want their commits to be signed by them and them only.
